gumroad/src/gumroad-block/block.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * BLOCK: gumroad-block
+ *
+ * Registering a basic block with Gutenberg.
+ * Simple block, renders and saves the same content without any interactivity.
+ */
+
+//  Import CSS.
+import './style.scss';
+import './editor.scss';
+
+const { __ } = wp.i18n; // Import __() from wp.i18n
+const { Component } = wp.element;
+const { registerBlockType } = wp.blocks; // Import registerBlockType() from wp.blocks
+const {
+  InspectorControls,
+  RichText
+} = wp.editor;
+
+const {
+  SelectControl,
+  PanelBody
+} = wp.components
+
+const gumURL = 'https://gum.co/';
+const defaultID = 'DviQY';
+
+class GumControls extends Component {
+  constructor( props ) {
+    super( ...arguments );
+  }
+
+  compose_url( value ) {
+    if (value == '') {
+      return gumURL + defaultID;
+    }
+    return gumURL + value;
+  }
+
+  render() {
+    const typeOptions = [
+      { value: 'none', label: __( 'Link' )},
+      { value: 'overlay', label: __( 'Overlay' )},
+      { value: 'embed', label: __( 'Embed' )}
+    ];
+
+    const buttonOptions = [
+      { value: '', label: __( 'Link' )},
+      { value: 'gumroad-button', label: __( 'Button' )}
+    ];
+
+    const wantedOptions = [
+      { value: '', label: __( 'unwanted' )},
+      { value: 'true', label: __( 'Wanted' )}
+    ];
+
+    const { setAttributes, attributes: { id, type, button, classes, wanted, url } } = this.props;
+
+    return(
+      <InspectorControls key="GumControls">
+        <PanelBody>
+          <label>Product ID:</label>
+          <RichText
+            tagName="div"
+            placeholder={ __( defaultID ) }
+            value={ id }
+            onChange={ ( value ) => {
+                setAttributes( { id: value } );
+                setAttributes( { url: this.compose_url( value ) } );
+              }
+            }
+          /><br />
+          <SelectControl
```

Comment:
we'll want text for the link text as well. "Link text"

Reply:
So the link text is directly manipulated in the gutenberg block rather than being manipulated in the settings sidebar. So placing a control here to change the text would be a bit... redundant. 

The default link text is currently `I want this!` Just like it is for the Shortcodes but we can change this to `Link text` if you prefer.

Reply:
Ah once I see ux that’ll be clearer. Was just guessing at func